File: 프로그래머스/0/181879. 길이에 따른 연산/길이에 따른 연산.java
class Solution {
    public int solution(int[] num_list) {
        int answer = 0;  // 합계용 변수
        int product = 1; // 곱셈용 변수
        
        // 조건에 따라 계산 방식 결정
        if (num_list.length >= 11) {
            // 11개 이상일 때는 모든 원소의 합을 구함
            for (int i = 0; i < num_list.length; i++) {
                answer += num_list[i];
            }
            return answer; // 합계 반환
        } else {
            // 10개 이하일 때는 모든 원소의 곱을 구함
            for (int i = 0; i < num_list.length; i++) {
                product *= num_list[i];
            }
            return product; // 곱 반환
        }
    }
}

File: 프로그래머스/0/181879. 길이에 따른 연산/길이에 따른 연산.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// num_list_len은 배열 num_list의 길이입니다.
int solution(int num_list[], size_t num_list_len) {
    int answer = 0;
    int sum = 1;
    if(num_list_len>=11){
         for(int i = 0; i<num_list_len; i++){
            answer+=num_list[i];
        }
    }
    else if(num_list_len<=10){
        for(int i = 0; i<num_list_len; i++){
            sum*=num_list[i];
        }
        answer = sum;
    }
   
    return answer;
}